packages/frontend/src/App.tsx

```diff
@@ -48,6 +49,11 @@ import { DashboardProvider } from './providers/DashboardProvider';
 import { TrackingProvider, TrackPage } from './providers/TrackingProvider';
 import { PageName } from './types/Events';
 
+// Keep original date time values on filters instead of converting dates into UTC when using JSON.stringify on API requests
+Date.prototype.toJSON = function () {
```

Comment:
Every time we do a request to the backend, we use JSON.stringify(query) (eg: useCompiledSql). 

This means that any Date in filters will be automatically converted to UTC, losing the timezone info. 

For example., if someone in the US has `Date('2022-10-01 00:00:00)` (a month filter) . Then it will be converted to `2022-09-31TXX:00:00Z`

This function is sending the current date without any UTC conversion

/home/rephus/Pictures/Screenshots/Screenshot from 2022-10-14 02-45-03.png